src/Microsoft.AspNetCore.SignalR.Common/Internal/Encoders/Base64Encoder.cs

```diff
@@ -23,18 +22,37 @@ public ReadOnlySpan<byte> Decode(byte[] payload)
             return decoded.Slice(0, written);
         }
 
+        private const int Int32OverflowLength = 10;
+
         public byte[] Encode(byte[] payload)
         {
-            Span<byte> buffer = new byte[Base64.GetMaxEncodedToUtf8Length(payload.Length)];
+            var maxEncodedLength = Base64.GetMaxEncodedToUtf8Length(payload.Length);
+
+            // Int32OverflowLength + 2 - Int32OverflowLength + length of separator (':') + length of terminator (';')
+            if (int.MaxValue - maxEncodedLength < Int32OverflowLength + 2)
+            {
+                throw new FormatException("The encoded message exceeds maximum supported size.");
+            }
+
+            //The format is: [{length}:{message};] so allocate enough to be able to write the entire message
+            Span<byte> buffer = new byte[Int32OverflowLength + 1 + maxEncodedLength + 1];
 
-            var status = Base64.EncodeToUtf8(payload, buffer, out _, out var written);
+            buffer[Int32OverflowLength] = (byte)':';
+            var status = Base64.EncodeToUtf8(payload, buffer.Slice(Int32OverflowLength + 1), out _, out var written);
             Debug.Assert(status == OperationStatus.Done);
 
-            using (var stream = new MemoryStream())
+            buffer.Slice(Int32OverflowLength + 1).Slice(written)[0] = (byte)';';
```

Comment:
This is the same as `buffer.Slice(Int32OverflowLength + 1 + written)`?

Reply:
I removed slicing altogether.